src/sentry/flags/endpoints/secrets.py

```diff
@@ -8,23 +11,69 @@
 from sentry.api.api_owners import ApiOwner
 from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import region_silo_endpoint
-from sentry.flags.endpoints import OrganizationFlagsEndpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import Serializer, register, serialize
 from sentry.flags.models import FlagWebHookSigningSecretModel
 from sentry.models.organization import Organization
 
 
+class FlagWebhookSigningSecretResponse(TypedDict):
+    createdAt: str
+    createdBy: int
+    id: int
+    provider: str
+    secret: str
+
+
+@register(FlagWebHookSigningSecretModel)
+class FlagWebhookSigningSecretSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs) -> FlagWebhookSigningSecretResponse:
+        secret = obj.secret
+        if len(secret) >= 10:
+            secret = secret[0:6] + "*" * len(secret[6:])
+        else:
+            secret = "*" * len(secret)
+
+        return {
+            "createdAt": obj.date_added.isoformat(),
+            "createdBy": obj.created_by,
+            "id": obj.id,
+            "provider": obj.provider,
+            "secret": secret,
+        }
+
+
 class FlagWebhookSigningSecretValidator(serializers.Serializer):
-    secret = serializers.CharField(required=True)
+    provider = serializers.ChoiceField(choices=[("launchdarkly", "launchdarkly")], required=True)
```

Comment:
dup launchdarkly

Reply:
Not a dupe just a bad API :\. Has to be this way...